packages/grid/x-grid/src/tests/rows.XGrid.test.tsx

```diff
@@ -384,6 +390,28 @@ describe('<XGrid /> - Rows', () => {
       expect(isVirtualized).to.equal(false);
     });
 
+    it('should allow defer rendering without race conditions', async () => {
```

Comment:
then let's get rid of it and keep the snap! Why do we need to waste time fighting the test framework when we can do a snap and the test is covered?

Reply:
Because the feedback loop for a snap is ~4 minutes when it's round 400ms for a test.

Reply:
@m4theushw Do you want to help with the test case?

Reply:
1. longer ie 350ms / 2sec ? => negligible in my opinion
2. It actually provides a much better assertion as it checks the full rendering of the grid.

But fine, I will fix the test if you prefer.

Reply:
Say you are working on a problem where fixing one thing, beak 3 others. Can you add an `.only` run the test and know which test fails or pass without looking? Or do you have to visual check that each instance is not broken, each time you iterate?

Looking closer, since it's a timing issue I'm not even sure it can be tested with the karma/jsdom test. What I would recommend is using the e2e test.

Reply:
1. I'm not sure I follow. You can open the story and fix the issue. You then check the other broken snap and fix them as well. In this particular case, it's the first rendering with a delay that we are testing so IMO snap is enough
2. Yes sure, I'm not saying we drop all the karma tests, I'm saying that for this case, we just take the frictionless path.

Reply:
It works in safari 13.2.1 where browserstack is failing.
![image](https://user-images.githubusercontent.com/936978/120810513-686b1780-c54b-11eb-84e7-f78e709b9429.png)

Reply:
> You can open the story and fix the issue. You then check the other broken snap and fix them as well.

1. Yes, this is the feedback loop, it will always be the same, for all the tests we right. My concern is with the **duration** of the feedback loop and how it can sustain a test codebase of x10 moe tests on the data grid. For this one to work, you mostly have to a. depending on the CI. b. check manually and visually if something is broken. This is my main problem with visual tests. On the other hand, with the programmatic test, the step in b. doesn't exist, you get the answer. the step in a. is what I would argue a lot faster. You can decide to only run the tests about the, say, filtering locally, and get the results in ~30s, not 6 minutes as in the CI.

For Safari, I have pushed on the e2e test direction in #1831 because we depend on having the timing right. We need to be async, we can't mock. 

Now, as I'm writing this, I'm not even sure that we need a e2e. I'm going to try to drop it. Just using a sleep 10ms.